--- origsrc/gcc-4.7.0/libstdc++-v3/python/Makefile.am	2012-02-21 04:55:54.000000000 -0600
+++ src/gcc-4.7.0/libstdc++-v3/python/Makefile.am	2012-06-13 01:36:03.073804100 -0500
@@ -26,7 +26,7 @@ include $(top_srcdir)/fragment.am
 if ENABLE_PYTHONDIR
 pythondir = $(prefix)/$(python_mod_dir)
 else
-pythondir = $(datadir)/gcc-$(gcc_version)/python
+pythondir = $(datadir)/gcc-$(gcc_version)/$(target_alias)/python
 endif
 
 all-local: gdb.py
--- origsrc/gcc-4.7.0/libstdc++-v3/python/Makefile.in	2012-02-17 14:26:32.000000000 -0600
+++ src/gcc-4.7.0/libstdc++-v3/python/Makefile.in	2012-06-13 01:36:31.667935100 -0500
@@ -300,7 +300,7 @@ WARN_CXXFLAGS = \
 
 # -I/-D flags to pass when compiling.
 AM_CPPFLAGS = $(GLIBCXX_INCLUDES)
-@ENABLE_PYTHONDIR_FALSE@pythondir = $(datadir)/gcc-$(gcc_version)/python
+@ENABLE_PYTHONDIR_FALSE@pythondir = $(datadir)/gcc-$(gcc_version)/$(target_alias)/python
 @ENABLE_PYTHONDIR_TRUE@pythondir = $(prefix)/$(python_mod_dir)
 nobase_python_DATA = \
     libstdcxx/v6/printers.py \
