--- origsrc/gcc-4.7.0/libjava/Makefile.am	2011-08-05 09:37:48.000000000 -0500
+++ src/gcc-4.7.0/libjava/Makefile.am	2012-06-12 02:57:16.508693300 -0500
@@ -1167,7 +1167,11 @@ endif
 else !NATIVE
 
 ecjx_LINK = $(GCJ_FOR_ECJX_LINK) $(ecjx_LDFLAGS)
+if INSTALL_ECJ_JAR
 ecjx_LDFLAGS = $(ECJX_BASE_FLAGS) $(ECJ_BUILD_JAR)
+else
+ecjx_LDFLAGS = $(ECJX_BASE_FLAGS) -Djava.class.path=$(ECJ_JAR)
+endif
 ecjx_LDADD = 
 ecjx_DEPENDENCIES = 
 ecjx.$(OBJEXT): ecjx.cc
--- origsrc/gcc-4.7.0/libjava/Makefile.in	2012-06-12 03:04:48.185527700 -0500
+++ src/gcc-4.7.0/libjava/Makefile.in	2012-03-22 02:37:39.000000000 -0500
@@ -8445,7 +8445,8 @@ ECJX_BASE_FLAGS = -findirect-dispatch \
 @NATIVE_TRUE@ecjx_LINK = $(GCJLINK) $(ecjx_LDFLAGS)
 @ENABLE_SHARED_FALSE@@NATIVE_TRUE@ecjx_LDFLAGS = $(ECJX_BASE_FLAGS) $(ECJ_BUILD_JAR) -fbootclasspath=$(BOOTCLASSPATH)
 @ENABLE_SHARED_TRUE@@NATIVE_TRUE@ecjx_LDFLAGS = $(ECJX_BASE_FLAGS) -Djava.class.path=$(ECJ_JAR)
-@NATIVE_FALSE@ecjx_LDFLAGS = $(ECJX_BASE_FLAGS) $(ECJ_BUILD_JAR)
+@INSTALL_ECJ_JAR_FALSE@@NATIVE_FALSE@ecjx_LDFLAGS = $(ECJX_BASE_FLAGS) -Djava.class.path=$(ECJ_JAR)
+@INSTALL_ECJ_JAR_TRUE@@NATIVE_FALSE@ecjx_LDFLAGS = $(ECJX_BASE_FLAGS) $(ECJ_BUILD_JAR)
 @NATIVE_FALSE@ecjx_LDADD = 
 @NATIVE_TRUE@ecjx_LDADD = -L$(here)/.libs $(extra_ldflags) \
 @NATIVE_TRUE@	$(am__append_28)
